### ISSUE TITLE
!VOTE does not work for the 11 categories

### ISSUE THREAD

**Issue by woctezuma** [chr233/ASFEnhance#169]
I have tried to vote for the 11 nominations:
```bash
!vote ASF 1868140,2296380,531640,1997040,1260320,1977170,1624540,1976950,1817230,1976950,2225070
```

My `MainBot` is fine, because I manually voted for it. However, the other bots have not cast a vote in each of the 11 categories.

```
>> !vote ASF 1868140,2296380,531640,1997040,1260320,1977170,1624540,1976950,1817230,1976950,2225070

<MainBot> Nominate at least 1 game: √
Nominate a game in each category: √
Play a game you nominated: √
Review or update your review of a game you nominated: √

<AltBot1> Nominate at least 1 game: √
Nominate a game in each category: ×
Play a game you nominated: ×
Review or update your review of a game you nominated: ×

<AltBot2> Nominate at least 1 game: √
Nominate a game in each category: ×
Play a game you nominated: ×
Review or update your review of a game you nominated: ×

<AltBot3> Nominate at least 1 game: √
Nominate a game in each category: ×
Play a game you nominated: ×
Review or update your review of a game you nominated: ×
```

This would probably be fixed by:
- https://github.com/chr233/ASFEnhance/pull/168

**Comment by woctezuma** [comment]
On a side-note, I have tried the following, yet it did not trigger `Play a game you nominated` for some reason.

```bash
!addlicense ASF a/2225070
!play ASF 2225070
!resume ASF
```

However, it worked for my `MainBot` yesterday. That is because the reason why it fails is due to the aforementioned issue.

**Comment by reanoff** [comment]
also marks 10 categories

**Comment by Botan626** [comment]
I confirm, command `vote bot1` doesn't vote in these categories:
![image](https://github.com/chr233/ASFEnhance/assets/28785698/a5777491-ce4e-400a-b11c-85568f5ea59d)
![image](https://github.com/chr233/ASFEnhance/assets/28785698/260c4506-3a2c-4bdc-b0c9-de225067ce1c)

**Comment by victoronlyarg** [comment]
I used the command with the apps id that you included without having seen the tasks before. Do you know if there is a way to delete the selected games?

**Comment by reanoff** [comment]
> I used the command with the apps id that you included without having seen the tasks before. Do you know if there is a way to delete the selected games?

just enter other appid

**Comment by chr233** [comment]
lol，i counted by 10, lost one category

**Comment by wulala1337** [comment]
> lol，i counted by 10, lost one category

更新了还是不行，还有2个类别就是不投票
![1](https://github.com/chr233/ASFEnhance/assets/120764089/fffeb298-9b33-45b9-a272-ee8412f546c4)
![2](https://github.com/chr233/ASFEnhance/assets/120764089/91b57717-fae5-4d86-934d-190d1a422287)

 - ASF 版本: 5.4.13.4 - 插件版本: 2.0.1.2 ========================================================================== { "EULA": true, "Statistic": true, "DevFeature": false, "DisabledCmds": null, "Address": null, "Addresses": [] } ========================================================================== - 错误类型: System.ArgumentNullException - 错误消息: Value cannot be null. (Parameter 'format') at System.ArgumentNullException.Throw(String) at System.String.FormatHelper(IFormatProvider provider, String format, ReadOnlySpan`1 args) at System.String.Format(String format, Object[] args) at ASFEnhance.Utils.AppendLineFormat(StringBuilder sb, String format, Object[] args) at ASFEnhance.Event.WebRequest.CheckSummerBadge(Bot bot) at ASFEnhance.Event.Command.ResponseSteamAwardVote(Bot bot, String gameIDs) at ArchiSteamFarm.Core.Utilities.InParallel[T](IEnumerable`1 tasks) at ASFEnhance.Event.Command.ResponseSteamAwardVote(String botNames, String choose) at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)

**Comment by chr233** [comment]
> > lol，i counted by 10, lost one category
> 
> 更新了还是不行，还有2个类别就是不投票
> ![1](https://github.com/chr233/ASFEnhance/assets/120764089/fffeb298-9b33-45b9-a272-ee8412f546c4)
> ![2](https://github.com/chr233/ASFEnhance/assets/120764089/91b57717-fae5-4d86-934d-190d1a422287)
> 
>  - ASF 版本: 5.4.13.4 - 插件版本: 2.0.1.2 ========================================================================== { "EULA": true, "Statistic": true, "DevFeature": false, "DisabledCmds": null, "Address": null, "Addresses": [] } ========================================================================== - 错误类型: System.ArgumentNullException - 错误消息: Value cannot be null. (Parameter 'format') at System.ArgumentNullException.Throw(String) at System.String.FormatHelper(IFormatProvider provider, String format, ReadOnlySpan`1 args) at System.String.Format(String format, Object[] args) at ASFEnhance.Utils.AppendLineFormat(StringBuilder sb, String format, Object[] args) at ASFEnhance.Event.WebRequest.CheckSummerBadge(Bot bot) at ASFEnhance.Event.Command.ResponseSteamAwardVote(Bot bot, String gameIDs) at ArchiSteamFarm.Core.Utilities.InParallel[T](IEnumerable`1 tasks) at ASFEnhance.Event.Command.ResponseSteamAwardVote(String botNames, String choose) at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)

还真是，杰出视觉风格好像就是投不上

**Comment by chr233** [comment]
> > lol，i counted by 10, lost one category
> 
> 更新了还是不行，还有2个类别就是不投票 ![1](https://user-images.githubusercontent.com/120764089/285095721-fffeb298-9b33-45b9-a272-ee8412f546c4.png) ![2](https://user-images.githubusercontent.com/120764089/285095726-91b57717-fae5-4d86-934d-190d1a422287.png)
> 
> * ASF 版本: 5.4.13.4 - 插件版本: 2.0.1.2 ========================================================================== { "EULA": true, "Statistic": true, "DevFeature": false, "DisabledCmds": null, "Address": null, "Addresses": [] } ========================================================================== - 错误类型: System.ArgumentNullException - 错误消息: Value cannot be null. (Parameter 'format') at System.ArgumentNullException.Throw(String) at System.String.FormatHelper(IFormatProvider provider, String format, ReadOnlySpan`1 args) at System.String.Format(String format, Object[] args) at ASFEnhance.Utils.AppendLineFormat(StringBuilder sb, String format, Object[] args) at ASFEnhance.Event.WebRequest.CheckSummerBadge(Bot bot) at ASFEnhance.Event.Command.ResponseSteamAwardVote(Bot bot, String gameIDs) at ArchiSteamFarm.Core.Utilities.InParallel[T](IEnumerable`1 tasks) at ASFEnhance.Event.Command.ResponseSteamAwardVote(String botNames, String choose) at ASFEnhance.ASFEnhance.OnBotCommand(Bot bot, EAccess access, String message, String[] args, UInt64 steamId)

![image](https://github.com/chr233/ASFEnhance/assets/13798576/50da69e2-20fa-4834-97a5-ccd7e8e0c35d)
我知道问题在哪里了，base64后的Protobuf有个 + 没有转义，得用url编码一下

**Comment by chr233** [comment]
fixed in 2.0.1.3

**Comment by wulala1337** [comment]
> fixed in 2.0.1.3
感谢更新！下午试了很多号投票都成功了，就是依然报错，但是不影响投票。

**Comment by chr233** [comment]
> > fixed in 2.0.1.3
> > 感谢更新！下午试了很多号投票都成功了，就是依然报错，但是不影响投票。

![image](https://github.com/chr233/ASFEnhance/assets/13798576/0d2235ee-aa03-43a2-8637-1696780d270f)
忘记添加翻译文本了……，稍后手动传一个版本上去

**Comment by chr233** [comment]
> > fixed in 2.0.1.3
> > 感谢更新！下午试了很多号投票都成功了，就是依然报错，但是不影响投票。

修好了，手动发了一个 fix 版本
